app/src/lib/stores/updates/changes-state.ts

```diff
@@ -96,40 +103,46 @@ export function updateChangedFiles(
 function getConflictState(
   status: IStatusResult,
   manualResolutions: Map<string, ManualConflictResolution>
-): IConflictState | null {
-  if (!status.mergeHeadFound) {
-    return null
+): ConflictState | null {
+  if (status.mergeHeadFound) {
+    const { currentBranch, currentTip } = status
+    if (currentBranch == null || currentTip == null) {
+      return null
+    }
+    return {
+      kind: 'merge',
+      currentBranch,
+      currentTip,
+      manualResolutions,
+    }
   }
 
-  const { currentBranch, currentTip } = status
-  if (currentBranch == null || currentTip == null) {
-    return null
-  }
+  if (status.rebaseContext !== null) {
+    const { currentTip } = status
+    if (currentTip == null) {
+      return null
+    }
 
-  return {
-    currentBranch,
-    currentTip,
-    manualResolutions,
+    const { targetBranch, originalBranchTip } = status.rebaseContext
+
+    return {
+      kind: 'rebase',
+      currentTip,
+      manualResolutions,
+      targetBranch,
+      originalBranchTip,
+    }
   }
+
+  return null
 }
 
-export function updateConflictState(
-  state: IChangesState,
+function performEffectsForMergeStateChange(
```

Comment:
i like this putting all the side effects into a single little function

Reply:
We've not been great with making side effects stand out from the rest of the code in the past, so hopefully being more explicit like this when we're in making changes will help keep the intent clear.